snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,159 @@
+                        Param("literal_1", LiteralMatch(None, AnyOptionalString())),
+                        self.uuid_column_pattern("_1"),
+                        self.formatted_uuid_pattern("_1"),
+                    )
+                ),
+            ),
+        )
+
+    def parse_uuid(self, lit: Expression) -> Expression:
+        if not isinstance(lit, Literal):
+            return lit
+
+        try:
+            parsed = uuid.UUID(lit.value)
+            return Literal(lit.alias, str(parsed))
+        except Exception:
+            return lit
+
+    def process_condition(self, exp: Expression) -> Expression:
+        result = self.uuid_in_condition.match(exp)
+        if result is not None:
+            new_params = []
+            if result.contains("formatted_uuid_column"):
+                alias = result.string("format_alias")
+                column = result.expression("formatted_uuid_column")
+                new_params.append(Column(alias, column.table_name, column.column_name,))
+            else:
+                new_params.append(result.expression("uuid_column"))
+
+            params_fn = result.expression("params")
+            new_fn_params = []
+            for param in params_fn.parameters:
+                new_fn_params.append(self.parse_uuid(param))
+
+            new_params.append(
+                FunctionCall(
+                    params_fn.alias, params_fn.function_name, tuple(new_fn_params)
+                )
+            )
+
+            return binary_condition(exp.alias, exp.function_name, *new_params)
+
+        result = self.uuid_condition.match(exp)
+        if result is not None:
+            new_params = []
+            for suffix in ["_0", "_1"]:
+                if result.contains("literal" + suffix):
+                    new_params.append(
+                        self.parse_uuid(result.expression("literal" + suffix))
+                    )
+                elif result.contains("uuid_column" + suffix):
+                    new_params.append(result.expression("uuid_column" + suffix))
+                elif result.contains("formatted_uuid_column" + suffix):
+                    alias = None
+                    if result.contains("format_alias" + suffix):
+                        alias = result.string("format_alias" + suffix)
+                    column = result.expression("formatted_uuid_column" + suffix)
+                    new_params.append(
+                        Column(alias, column.table_name, column.column_name,)
+                    )
+
+            return binary_condition(exp.alias, exp.function_name, *new_params)
+
+        return exp
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        condition = query.get_condition_from_ast()
+        if condition:
+            query.set_ast_condition(None)
+            for cond in condition:
+                if is_condition(cond):
+                    query.add_condition_to_ast(self.process_condition(cond))
```

Comment:
I am not sure this works for nested conditions, like:
```
AND
   a = b
   OR 
      b = c
      c = d
```

`for cond in condition` would traverse the tree in postorder, then transform each condition and add it back to the condition. The result would be 

```
AND
   c = d # transformed
   AND 
      b = c. # transformed
      AND
          c = d # transformed
         AND # the existing condition.
            a = b
            OR 
                b = c
                c = d
```


I think what you want to do is:
```
condition = query.get_ast_condition()
if conndition is not None:
       query.set_ast_condition(condition.transform(self.process_condition)
```